src/main/java/com/sap/sgs/phosphor/fosstars/data/github/VendorDataMatcher.java

```diff
@@ -0,0 +1,131 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import com.sap.sgs.phosphor.fosstars.nvd.Matcher;
+import com.sap.sgs.phosphor.fosstars.nvd.data.Affects;
+import com.sap.sgs.phosphor.fosstars.nvd.data.CVE;
+import com.sap.sgs.phosphor.fosstars.nvd.data.CVEDataMeta;
+import com.sap.sgs.phosphor.fosstars.nvd.data.NvdEntry;
+import com.sap.sgs.phosphor.fosstars.nvd.data.ProductData;
+import com.sap.sgs.phosphor.fosstars.nvd.data.Vendor;
+import com.sap.sgs.phosphor.fosstars.nvd.data.VendorData;
+import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProject;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This is a matcher that check if vendor data from an NVD entry matched to
+ * a project's owner and name.
+ */
+public class VendorDataMatcher implements Matcher {
+
+  /**
+   * A logger.
+   */
+  private static final Logger LOGGER = LogManager.getLogger(VendorDataMatcher.class);
+
+  /**
+   * Creates a new matcher for a project.
+   *
+   * @param project The project.
+   * @return The new matcher.
+   */
+  public static VendorDataMatcher with(GitHubProject project) {
+    return new VendorDataMatcher(project);
+  }
+
+  /**
+   * A project to be checked.
+   */
+  private final GitHubProject project;
+
+  /**
+   * Initializes a new matcher.
+   *
+   * @param project A project to be checked.
+   */
+  private VendorDataMatcher(GitHubProject project) {
+    this.project = Objects.requireNonNull(project, "Null is not a project!");
+  }
+
+  @Override
+  public boolean match(NvdEntry entry) {
+    Objects.requireNonNull(entry, "NVD entry can't be null!");
+
+    CVE cve = entry.getCve();
+    if (cve == null) {
+      LOGGER.warn("No CVE in NVD entry");
+      return false;
+    }
+
+    CVEDataMeta meta = cve.getCveDataMeta();
+    if (meta == null) {
+      LOGGER.warn("No metadata in NVD entry");
+      return false;
+    }
+    String cveId = meta.getID();
+    if (cveId == null) {
+      LOGGER.warn("No CVE ID in NVD entry");
+      cveId = "unknown";
+    }
+
+    Affects affects = cve.getAffects();
+    if (affects == null) {
+      LOGGER.warn("No affects in NVD entry for {}", cveId);
+      return false;
+    }
+
+    Vendor cveVendor = affects.getVendor();
+    if (cveVendor == null) {
+      LOGGER.warn("No vendor in NVD entry for {}", cveId);
+      return false;
+    }
+
+    for (VendorData vendorData : cveVendor.getVendorData()) {
+      if (match(vendorData)) {
```

Comment:
I think you can just return this if predicate directly. It can return false also.

Reply:
I am not sure if I understand what you mean.

Reply:
return `match(vendorData);`

Reply:
There may be multiple `VendorData`. If first one doesn't match, it has to check the next one.